### ISSUE TITLE
 redefinition of  ARMCI_Init_mpi_comm armci/src/common/capi.c

### ISSUE THREAD

**Issue by edoapra** [GlobalArrays/ga#292]
` ARMCI_Init_mpi_comm` in armci/src/common/capi.c 
https://github.com/GlobalArrays/ga/blob/380ce3020c667ae398f116e8619d35dd9bce9f2d/armci/src/common/capi.c#L204
https://github.com/GlobalArrays/ga/blob/380ce3020c667ae398f116e8619d35dd9bce9f2d/armci/src/common/capi.c#L214

This results in a fatal error with icc from OneAPI 2022.2.1

see 
https://github.com/GlobalArrays/ga/actions/runs/3636154648/jobs/6135877178#step:7:3827

**Comment by bjpalmer** [comment]
This should be fixed. It must have snuck in when merging the init_comm branch with develop.

**Comment by edoapra** [comment]
> This should be fixed. It must have snuck in when merging the init_comm branch with develop.

Now the actions stops with a different error.
The source code of `ga++/testing/create_irreg_test.cc` seems to be missing
```
make[1]: *** No rule to make target 'ga++/testing/create_irreg_test.cc', needed by 'ga++/testing/create_irreg_test.o'.  Stop.
make[1]: *** Waiting for unfinished jobs.
```
https://github.com/GlobalArrays/ga/actions/runs/3650220544/jobs/6165934851

**Comment by bjpalmer** [comment]
Oops, forgot to add that to the build. Actually, it's a pretty crappy test. Probably would be better to remove it entirely.

**Comment by ajaypanyala** [comment]
Fixed via [3777e30](https://github.com/GlobalArrays/ga/commit/3777e302c2102c67741eeb2cf19ab435d4a5746d) and [b290b7c](https://github.com/GlobalArrays/ga/commit/b290b7c0d68abee010e0759eaf6f667dda5d16bb)
